### PR TITLE
[Release 2.12.0] Consolidated release notes

### PR DESCRIPTION
**Pull request by prudhvigodithi**
### Description
[Release 2.12.0] Consolidated release notes

### Issues Resolved
Part of https://github.com/opensearch-project/opensearch-build/issues/4115

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-2.12.0.md [hunk 1]
```diff
@@ -0,0 +1,1306 @@
...
+<li>A new agent framework added to ML Commons uses remote large language learning models (LLMs) for step-by-step problem-solving and can coordinate machine learning tools using LLMs. The framework includes a flow agent and a conversational agent in this release.</li>
+<li>Users can now query multiple clusters with cross-cluster monitors through the Alerting plugin.</li>
+</ul>
+
+<h2>DEPRECATION NOTICE</h2>
+
+<h3>CentOS7</h3>
+<ul>
+Please note that OpenSearch will deprecate support for CentOS Linux 7 as a continuous integration build image and supported operating system in an upcoming release, as CentOS Linux 7 will reach end-of-life on June 30, 2024 (see <a href="https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/">#notice</a>). To view OpenSearch's compatible operating systems, visit <a href="https://blog.centos.org/2023/04/end-dates-are-coming-for-centos-stream-8-and-centos-linux-7/">#operating-system-compatibility</a>.
```
Comment (jhmcintyre) [inline release-notes/opensearch-release-notes-2.12.0.md:35]: the first link looks right, but the second link should be replaced with: https://opensearch.org/docs/latest/install-and-configure/install-opensearch/index/#operating-system-compatibility..

File: release-notes/opensearch-release-notes-2.12.0.md [hunk 2]
```diff
@@ -0,0 +1,1306 @@
+<h1>OpenSearch and OpenSearch Dashboards 2.12.0 Release Notes</h1>
+<h2>RELEASE HIGHLIGHTS</h2>
+<ul>
+OpenSearch 2.12.0 includes a number of features designed to increase performance for search and analytics applications as well as user experience enhancements and several new and enhanced machine learning tools. Experimental functionality includes the OpenSearch Assistant Toolkit for building interactive user experiences and new cross-cluster monitors. OpenSearch upgrades to Apache Lucene 9.9.1 with this release.
```
Comment (sohami) [inline release-notes/opensearch-release-notes-2.12.0.md:4]: lucene version is 9.9.2. Ref [here](https://github.com/opensearch-project/OpenSearch/blob/2.12/buildSrc/version.properties#L2)

**Comment by nateynateynate** [comment]
@prudhvigodithi I tried to push a change to your PR but I didn't have access. I think you'll have to update this one on your own. One link and two references to Lucene 9.1.1.

File: release-notes/opensearch-release-notes-2.12.0.md (hunk 1 above)
Comment (prudhvigodithi) [reply release-notes/opensearch-release-notes-2.12.0.md]: Updated the link in my latest commit.

File: release-notes/opensearch-release-notes-2.12.0.md (hunk 2 above)
Comment (prudhvigodithi) [reply release-notes/opensearch-release-notes-2.12.0.md]: Updated to 9.9.2.

**Review by bbarani — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]

File: release-notes/opensearch-release-notes-2.12.0.md [hunk 3]
```diff
@@ -0,0 +1,1306 @@
...
+<h2>RELEASE HIGHLIGHTS</h2>
+<ul>
+OpenSearch 2.12.0 includes a number of features designed to increase performance for search and analytics applications as well as user experience enhancements and several new and enhanced machine learning tools. Experimental functionality includes the OpenSearch Assistant Toolkit for building interactive user experiences and new cross-cluster monitors. OpenSearch upgrades to Apache Lucene 9.9.2 with this release. 
+</ul>
+
+<h2>NEW FEATURES</h2>
+<ul>
+<li>Integration with Apache Spark lets you analyze all of your operational data in a single place using OpenSearch in combination with Apache Spark. </li>
+<li>Conversational search is generally available, providing comprehensive functionality to build conversational experiences using OpenSearch’s lexical, vector, and hybrid search features.</li>
+<li>A new workflow engine lets you automate configurations for ML Commons resources, allowing you to set up machine learning resources to support AI use cases without the need to manually create resources or write custom scripts.</li>
```
Comment (jhmcintyre) [inline release-notes/opensearch-release-notes-2.12.0.md:11]: This item (new workflow engine) should be moved to the Experimental section below. Per docs, now seeing that this feature is experimental. https://opensearch.org/docs/latest/automating-configurations/index/

**Pull request merged**